### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by johnnyshields**
- Use consistent header sizes
- Add Version Support section
- Improve structure by adding "Command Line Usage" and "AWS Configuration" section headers
- Add aliases (e.g. `-f`) for all command line switches
- Remove redundant text in some places

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,15 +57,15 @@ end
...
+#### Configuration Defaults
 
 The following defaults apply to the configuration items:
 
 | Setting                    | Default                            |
 | -----------------          | ----------------------------       |
-| aws_access_key_id          | ```ENV['AWS_ACCESS_KEY_ID']```     |
-| aws_secret_access_key      | ```ENV['AWS_SECRET_ACCESS_KEY']``` |
-| bucket                     | ```ENV['AWS_BUCKET']```            |
+| aws_access_key_id          | -                                  |
```
Comment (fredjean) [inline README.md]: S3 Sync (and fog) both support the use of environment variables to pass in the AWS credentials. They are effectively the default for s3_sync. We should continue to mention this in the README.

File: README.md [hunk 2]
```diff
@@ -123,13 +125,7 @@ method or passed through the ```.s3_sync``` configuration file.
 
 Alternatively, if you are running builds on EC2 instance which has approrpiate IAM role, then you don't need to think about specifying credentials at all – they will be pulled from AWS metadata service.
 
-#### Through the command line
-
-The aws credentials can also be passed via a command line options
-`--aws_access_key_id` and `aws_secret_access_key`. They should override
```
Comment (fredjean) [inline README.md]: Nit, it should be `--aws_secret_access_key`

**Comment by fredjean** [comment]
Thank you very much for your help. Just a couple nits for this PR.

File: README.md (hunk 1 above)
Comment (johnnyshields) [inline README.md]: The default value of the configuration (without doing anything) is nil. ENV vars and Thor args are documented separately.

Side note: I could not get the ENV vars to work from command line. How does one pass in ENV vars with Thor?

File: README.md (hunk 2 above)
Comment (johnnyshields) [inline README.md]: You are commenting on a deleted line (red). If you look at my PR changes (green) you'll see that I changed it to --aws_secret_access_key

Comment (fredjean) [inline README.md]: My wrong... I wasn't fully caffeinated.

Johnny Shields wrote:

File: README.md (hunk 1 above)
Comment (fredjean) [inline README.md]: That's a fair point.

The usual CLI tricks do work for environment variables:

```
$ export AWS_BUCKET=a-bucket.com
```

or 

```
$ AWS_BUCKET=a-bucket.com middleman s3_sync
```

I tend to add the `middleman-dotenv` gem to my `config.rb` file.

Most CI services do provide the ability to set environment variables for a build, which helps keep credentials out of the repos.

Comment (johnnyshields) [inline README.md]: Hmm ok. Well, unless you feel strongly I think it's better to indicate these vars are nil by default, i.e. the user must set them in some way.

Comment (fredjean) [inline README.md]: I'm convinced. I'll merge.

**Pull request merged**